tutorials/llm/llama-3/nemo2-sft-peft/nemo2-sft.ipynb

```diff
@@ -0,0 +1,632 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### AutoResume\n",
+    "In NeMo 2.0 we can directly pass in Llama3-8b's Hugging Face ID to start SFT without manually converting it into NeMo checkpoint format like in NeMo 1.0."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def resume() -> run.Config[nl.AutoResume]:\n",
+    "    return run.Config(\n",
+    "        nl.AutoResume,\n",
+    "        restore_config=run.Config(nl.RestoreConfig,\n",
+    "            path=\"nemo://meta-llama/Meta-Llama-3-8B\"\n",
+    "        ),\n",
+    "        resume_if_exists=True,\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "### NeMo 2.0 finetun API\n",
+    "Using all the components we created above, we can call NeMo 2.0 finetun API:\n",
+    "```\n",
+    "llm.finetune(\n",
+    "    model=llama3_8b(),\n",
+    "    data=dolly(),\n",
+    "    trainer=trainer(),\n",
+    "    log=logger(),\n",
+    "    optim=adam_with_cosine_annealing(),\n",
+    "    resume=resume(),\n",
+    ")\n",
+    "```\n",
+    "Below is a python script that you can save as a file e.g. `nemo2-sft.py`, and run SFT training, using all components we created above and NeMo 2.0 finetune API. The script cannot be directly executed in interactive environment like a notebook. We can execute by `torchrun --nproc_per_node=<NUM_GPU> nemo2-sft.py` when multiple GPU is used."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def configure_finetuning_recipe():\n",
+    "    return run.Partial(\n",
+    "        llm.finetune,\n",
+    "        model=llama3_8b(),\n",
+    "        trainer=trainer(),\n",
+    "        data=dolly(),\n",
+    "        log=logger(),\n",
+    "        optim=adam_with_cosine_annealing(),\n",
+    "        resume=resume(),\n",
+    "    )\n",
+    "\n",
+    "\n",
+    "def local_executor_torchrun(nodes: int = 1, devices: int = 2) -> run.LocalExecutor:\n",
+    "    # Env vars for jobs are configured here\n",
+    "    env_vars = {\n",
+    "        \"TORCH_NCCL_AVOID_RECORD_STREAMS\": \"1\",\n",
+    "        \"NCCL_NVLS_ENABLE\": \"0\",\n",
+    "        \"NVTE_DP_AMAX_REDUCE_INTERVAL\": \"0\",\n",
+    "        \"NVTE_ASYNC_AMAX_REDUCTION\": \"1\",\n",
+    "        \"NVTE_FUSED_ATTN\": \"0\",\n",
+    "    }\n",
+    "\n",
+    "    executor = run.LocalExecutor(ntasks_per_node=devices, launcher=\"torchrun\", env_vars=env_vars)\n",
+    "\n",
+    "    return executor\n",
+    "\n",
+    "if __name__ == '__main__':\n",
+    "    run.run(configure_finetuning_recipe(), executor=local_executor_torchrun())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 4 Evaluation\n",
```

Comment:
revise heading and change to imperative verb.

"## Step 4. Generate Results from Trained PEFT Checkpoints \n",